spring-cloud-gcp-kms/src/main/java/com/google/cloud/spring/kms/KmsTemplate.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.kms;
+
+import java.util.Base64;
+
+import com.google.cloud.kms.v1.CryptoKeyName;
+import com.google.cloud.kms.v1.DecryptRequest;
+import com.google.cloud.kms.v1.DecryptResponse;
+import com.google.cloud.kms.v1.EncryptRequest;
+import com.google.cloud.kms.v1.EncryptResponse;
+import com.google.cloud.kms.v1.KeyManagementServiceClient;
+import com.google.cloud.spring.core.GcpProjectIdProvider;
+import com.google.common.hash.Hashing;
```

Comment:
@meltsufin - Is it ok to use Guava here? Looks like @gkatzioura needed it to verify crc32c; looked around and this looks like the most convenient library to use in java 8.

Reply:
I think it's fine, especially since it's the recommended approach in our documentation.